synapse/federation/federation_server.py

```diff
@@ -727,27 +727,6 @@ async def _handle_received_pdu(self, origin: str, pdu: EventBase) -> None:
             if the event was unacceptable for any other reason (eg, too large,
             too many prev_events, couldn't find the prev_events)
         """
-        # check that it's actually being sent from a valid destination to
-        # workaround bug #1753 in 0.18.5 and 0.18.6
-        if origin != get_domain_from_id(pdu.sender):
```

Comment:
This needs to get removed as we're now sending events in transactions that aren't from the originating server. It's safe to remove due to the bug having long since been fixed.